Lib/test/test_sqlite3/test_cli.py

```diff
@@ -200,5 +205,74 @@ def test_color(self):
             self.assertIn('\x1b[1;35mOperationalError (SQLITE_ERROR)\x1b[0m: '
                           '\x1b[35mnear "sel": syntax error\x1b[0m', err)
 
+@requires_subprocess()
+class CompletionTest(unittest.TestCase):
+    PS1 = "sqlite> "
+
+    @classmethod
+    def setUpClass(cls):
+        readline = import_module("readline")
+        if readline.backend == "editline":
+            raise unittest.SkipTest("libedit readline is not supported")
+
+    def test_keyword_completion(self):
+        script = textwrap.dedent("""
+            import readline
+            readline.parse_and_bind("set colored-completion-prefix off")
+            from sqlite3.__main__ import main; main()
+        """)
+        # List candidates starting with 'S', there should be multiple matches.
+        input = b"S\t\tEL\t 1;\n.quit\n"
+        output = run_pty(script, input)
+        self.assertIn(b"SELECT", output)
+        self.assertIn(b"SET", output)
+        self.assertIn(b"SAVEPOINT", output)
+        self.assertIn(b"(1,)", output)
+
+        # Keywords are completed in upper case for even lower case user input
+        input = b"sel\t\t 1;\n.quit\n"
+        output = run_pty(script, input)
+        self.assertIn(b"SELECT", output)
+        self.assertIn(b"(1,)", output)
+
+    def test_nothing_to_complete(self):
+        script = textwrap.dedent("""
+            import readline
+            readline.parse_and_bind("set colored-completion-prefix off")
+            from sqlite3.__main__ import main; main()
+        """)
+        input = b"xyzzy\t\t\b\b\b\b\b.quit\n"
+        output = run_pty(script, input, env={"NO_COLOR": "1"})
+        output_lines = output.decode().splitlines()
+        line_num = next(i for i, line in enumerate(output_lines, 1)
+                        if line.startswith(f"{self.PS1}xyzzy"))
```

Comment:
```suggestion
        line_num = next(i for i, line in enumerate(output_lines, 1)
                        if line.startswith(f"{self.PS1}xyzzy"), -1)
```

Just to prevent some possible StopIteration if the test fails (easier to check that line_num == -1 indicating an incorrect line number)